### PR TITLE
if referenced attribute is object update current reference or create n…

### PR DESCRIPTION
**Pull request by danielmurray**
…ew reference model

### CODE REVIEW

File: dist/js/orb.js [hunk 1]
```diff
@@ -623,9 +623,13 @@ require('./queries');
...
                     delete attributes[attribute];
 
-                    if (value instanceof Backbone.Model) {
-                        self.references[attribute] = value;
-                        if (field) {
+                    if (value instanceof Object ) {
+                        if(value instanceof Backbone.Model){
+                          self.references[attribute] = value;
+                        }else{
+                          self.references[attribute].set(value);
```
Comment (erichulser) [inline dist/js/orb.js]: what type of object is this?  When would this scenario arise?

File: src/model.js [hunk 2]
```diff
@@ -284,17 +284,25 @@
...
-                        self.references[attribute] = value;
-                        if (field) {
+                    if (value instanceof Object ) {
+                        if(value instanceof Backbone.Model){
+                            self.references[attribute] = value;
+                        } else if(self.references[attribute]) {
+                            self.references[attribute].set(value);
+                        } else if (model) {
+                            self.references[attribute] = new model(value);
+                        }
```
Comment (erichulser) [inline src/model.js:308]: in this last scenario, nothing happens (if no model is defined)

**Comment by erichulser** [comment]
Has a merge conflict...

File: src/model.js (hunk 2 above)
Comment (danielmurray) [inline src/model.js:308]: i don't see how that would be an error, if something is defined as a reference on the model, shouldn't it be guaranteed to have a model?

**Pull request merged**